### PR TITLE
Clean up subscript-sized int definitions

### PR DESCRIPTION
**Pull request by klausler**
After some discussion with Tim, define a couple of typedef aliases for std::int64_t, use them, and remove the means by which large array address calculations could be disabled.

### CODE REVIEW

File: lib/evaluate/fold.cc [hunk 1]
```diff
@@ -966,8 +966,8 @@ static std::optional<Constant<SubscriptInteger>> GetConstantSubscript(
             }
           },
           [](Triplet &triplet) -> std::optional<Constant<SubscriptInteger>> {
-            std::optional<std::int64_t> lbi{1}, ubi;
-            std::optional<std::int64_t> stride{ToInt64(triplet.stride())};
+            std::optional<ConstantSubscript> lbi{1}, ubi;
+            std::optional<ConstantSubscript> stride{ToInt64(triplet.stride())};
```
Comment (tskeith) [inline lib/evaluate/fold.cc]: Using `ToInt64` here seems to break the abstraction. If `ConstantSubscript` were a wider type than `std::int64_t` we wouldn't get all of the bits.

Ideally we would want `std::optional<ConstantSubscript> ToConstantSubscript(...)`, as well as `ToLengthCIntType()` and `ToSubscriptCIntType()`, but that seems excessive.

Comment (klausler) [reply lib/evaluate/fold.cc]: Tell me what you want.

Comment (tskeith) [reply lib/evaluate/fold.cc]: What do you think of changing `ToInt64` to `std::optional<std::intmax_t> ToIntMax(...)`? Then we know whatever `SubscriptCIntType` is, we won't lose precision in this kind of initialization.

If you don't think that is worth the trouble, this is fine as it is.

Comment (klausler) [reply lib/evaluate/fold.cc]: Braced initialization would become errors as those 128-bit integer values were used to initialize `ConstantSubscript` values that were 64 bits.  So I'd have to put casts in, or an intervening conversion function call, or change `ConstantSubscript` to be 128 bits.

**Comment by klausler** [comment]
Changing names to the preferred `ConstantSubscript` so that the same name can be used everywhere.

**Comment by klausler** [comment]
I updated the change; please take another look whenever convenient.

**Comment by tskeith** [comment]
> I updated the change; please take another look whenever convenient.

Looks good to me.

**Pull request merged**